src/main/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/api/controllers/advice/ControllerAdvice.kt

```diff
@@ -47,6 +49,13 @@ class ControllerAdvice {
     return ResponseEntity(ErrorResponse(status = 403, developerMessage = e.message), HttpStatus.FORBIDDEN)
   }
 
+  @ExceptionHandler(DuplicateSourceRecordFound::class)
+  @ResponseStatus(HttpStatus.CONFLICT)
+  fun handle(e: DuplicateSourceRecordFound): ResponseEntity<SupplementaryRiskDto?> {
```

Comment:
This would be better if it was generic rather than specific to SupplementaryRiskDto